graylog2-server/src/main/java/org/graylog/events/processor/EventProcessorDependencyCheck.java

```diff
@@ -62,14 +65,10 @@ public boolean canProcessTimerange(DateTime maxTimestamp, Set<String> processorD
      * Caveat: This only looks at the processing status and doesn't take the Elasticsearch {@code index.refresh_interval}
      * into account!
      *
-     * @param latestTimestamp the timestamp to check
+     * @param timeRange the timestamp to check
      * @return true if messages up to the given latestTimestamp have already been indexed, false otherwise
      */
-    public boolean hasMessagesIndexedUpTo(DateTime latestTimestamp) {
-        // If there is no timestamp returned, there is no node in the cluster that is processing any messages.
-        // In that case we return false because there is nothing to process.
-        return processingStatusService.earliestPostIndexingTimestamp()
-                .map(latestPostIndexTimestamp -> latestPostIndexTimestamp.isAfter(latestTimestamp) || latestPostIndexTimestamp.isEqual(latestTimestamp))
-                .orElse(false);
+    public boolean hasMessagesIndexedUpTo(TimeRange timeRange) {
```

Comment:
Change this signature would need another PR for the new SigmaEventProcessor.
But do we need that? (see below)

Reply:
https://github.com/Graylog2/graylog-plugin-enterprise/pull/4194